client/src/pages/addFlight/index.js

```diff
@@ -9,29 +10,101 @@ import {
   Header,
   SideNav
 } from "../../components";
+
 import "./addflight.css";
 
-class UpdateFlight extends Component {
+class AddFlight extends Component {
   state = {
-    selectValue: ""
+    form: {},
+    selectedCity: {},
+    flightDuration: "",
+    succesfullyAdded: false,
+    alert: null
   };
 
   handleAddFlight = e => {
     e.preventDefault();
-    console.log("on added flights will be handled here");
+    const { dateValue, airCraftTypeValue } = this.state.form;
+    const { originValue, destinationValue } = this.state.selectedCity;
+
+    const data = JSON.stringify({
+      originValue,
+      destinationValue,
+      dateValue,
+      airCraftTypeValue
+    });
+    fetch("/addFlight_post", {
+      credentials: "same-origin",
+      headers: {
+        "content-type": "application/json"
+      },
+      method: "POST",
+      body: data
+    })
+      .then(response => response.json(data))
+      .then(data => {
+        if (data.succesfullyAdded) {
+          this.setState({
+            succesfullyAdded: true
+          });
+          this.handleSuceessAddFlight();
+        } else {
+          this.setState({
+            succesfullyAdded: false
+          });
+        }
+      })
+      .catch(err => {
+        console.log("There has been an error ", err);
+      });
   };
 
-  handleSelectChange = value => {
-    console.log(value);
-    this.setState({ selectValue: value });
+  handleSuceessAddFlight() {
+    const getAlert = () => (
+      <SweetAlert
+        success
+        title="Done!"
+        confirmBtnBsStyle="success"
+        cancelBtnBsStyle="success"
+        onConfirm={() => this.hideAlert()}
+      >
+        New Flight is Added
+      </SweetAlert>
+    );
+
+    this.setState({
+      alert: getAlert()
+    });
+  }
+
+  hideAlert() {
+    this.setState({
+      alert: null
+    });
+  }
+
+  handleSelectChange = e => {
+    const { selectedCity } = this.state;
+    const updatedSelectedCity = { ...selectedCity };
+    updatedSelectedCity[e.target.name] = e.target.value;
+    this.setState({ selectedCity: updatedSelectedCity });
   };
 
   calculateTime = e => {
-    // e.preventDefault();
-    console.log("Flight time calcuated");
+    e.preventDefault();
+    const randomNo = Math.floor(Math.random() * 11);
```

Comment:
Moment.js might help when you're ready to calculate real time durations:

https://momentjs.com/
http://momentjs.com/docs/#/durations/diffing/